### PR TITLE
update docs, node, add docker & bash linting

### PR DESCRIPTION
**Pull request by sammcj**
- Add minimal PR template
- Update docs to pin checkout action to v3
- Update node docker image from 12 to 16 (Fixes https://github.com/anothrNick/github-tag-action/issues/172)
- Add bash linting (on PR) (Fixes https://github.com/anothrNick/github-tag-action/issues/171)
- Add dockerfile linting (on PR)
- Add Github Actions linting (on PR)
- Update apk install to best practice --no-cache option

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM node:12-alpine3.15
+FROM node:16-alpine
 LABEL "repository"="https://github.com/anothrNick/github-tag-action"
```
Comment (sbe-arg) [inline Dockerfile:2]: Have you tested this locally?

Comment (sammcj) [reply Dockerfile:2]: I've built it without issue, but I'm not sure that there is a way to test it properly locally?

It relies on running on Github, for example in the entrypoint there are github-only variables such as $GITHUB_WORKSPACE, $source etc....

relates to #174

Comment (sbe-arg) [reply Dockerfile:2]: Let me test this

Comment (sammcj) [reply Dockerfile:2]: Thanks @sbe-arg , if let me know your local test process, I'll add those instructions to the readme as part of this PR as well!

**Review by sbe-arg — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (sbe-arg) [reply Dockerfile:2]: The simplest way is to mock it from a branch.

But I do try to test that semver works as expected in the new node version with 
```
nvm use 16
npm install -g semver
semver -i minor 0.0.0
semver -i major v0.0.0
```

https://github.com/sbe-arg/testing-tags/blob/master/.github/workflows/main.yml#L16

https://github.com/sbe-arg/testing-tags/runs/8297114662?check_suite_focus=true

**Pull request merged**